Source/WebCore/PAL/pal/text/TextCodecCJK.cpp

```diff
@@ -273,7 +273,7 @@ static Vector<uint8_t> eucJPEncode(StringView string, Function<void(char32_t, Ve
     result.reserveInitialCapacity(string.length());
 
     auto characters = string.upconvertedCharacters();
-    for (WTF::CodePointIterator<UChar> iterator(characters.get(), characters.get() + string.length()); !iterator.atEnd(); ++iterator) {
+    for (WTF::CodePointIterator<UChar> iterator({ characters.get(), string.length() }); !iterator.atEnd(); ++iterator) {
```

Comment:
Should `upconvertedCharacters` turn itself into a `span` instead of a pointer or in addition to a pointer?

Reply:
Yes, I think we need to update upconvertedCharacters. It currently only has a pointer and doesn't know about its length. The length has to come from somewhere else (usually the original string), which is too error prone. I have an item on my TODO list to look into it.

Reply:
Since it’s only used in cases that we don’t want to optimize for performance it should be an easy change. It’s intended to be a convenience to let you handle cases where we always want to use `char16_t` without having to write out buffer management code. The reason it only produced a pointer was the kind of idiom it was designed for. You’d only call it right where you’d have to pass a pointer to something that needs UTF-16. The original idea was that if the code is performance critical we can always use more efficient idioms that are more verbose.

Reply:
Ok, I have a patch ready for upconvertedCharacters but I'll perf A/B test it just in case.